torch_harmonics/examples/models/_layers.py

```diff
@@ -258,123 +250,19 @@ def __init__(self,
 
         if self.operator_type == "diagonal":
             weight_shape += [self.modes_lat, self.modes_lon]
-            from .contractions import contract_diagonal as _contract
+            self.contract_func = "...ilm,oilm->...olm"
         elif self.operator_type == "block-diagonal":
             weight_shape += [self.modes_lat, self.modes_lon, self.modes_lon]
-            from .contractions import contract_blockdiag as _contract
+            self.contract_func = "...ilm,oilnm->...oln"
         elif self.operator_type == "driscoll-healy":
             weight_shape += [self.modes_lat]
-            from .contractions import contract_dhconv as _contract
+            self.contract_func = "...ilm,oil->...olm"
         else:
             raise NotImplementedError(f"Unkonw operator type f{self.operator_type}")
 
         # form weight tensors
-        scale = math.sqrt(gain / in_channels) * torch.ones(self.modes_lat, 2)
-        scale[0] *=  math.sqrt(2)
-        self.weight = nn.Parameter(scale * torch.view_as_real(torch.randn(*weight_shape, dtype=torch.complex64)))
-
-        # get the right contraction function
-        self._contract = _contract
-
-        if bias:
-            self.bias = nn.Parameter(torch.zeros(1, out_channels, 1, 1))
-
-
-    def forward(self, x):
-
-        dtype = x.dtype
-        x = x.float()
-        residual = x
-
-        with torch.autocast(device_type="cuda", enabled=False):
-            x = self.forward_transform(x)
-            if self.scale_residual:
-                residual = self.inverse_transform(x)
-
-
-        x = torch.view_as_real(x)
-        x = self._contract(x, self.weight)
-        x = torch.view_as_complex(x)
-
-        with torch.autocast(device_type="cuda", enabled=False):
-            x = self.inverse_transform(x)
-
-        if hasattr(self, "bias"):
-            x = x + self.bias
-        x = x.type(dtype)
-
-        return x, residual
-
-class FactorizedSpectralConvS2(nn.Module):
```

Comment:
is this completely gone?

Reply:
yes - this is to simplify example architectures and remove dependencies